lib/remote-model.js

```diff
@@ -1,120 +1,96 @@
+const requestLib = require('request');
 const _ = require('underscore');
-
+const hmpoLogger = require('hmpo-logger');
 
 class RemoteModel extends LocalModel {
-    fetch(options, callback) {
-        if (typeof options === 'function' && arguments.length === 1) {
-            callback = options;
-            options = {};
-        } else if (!options) {
-            options = {};
-        }
-        let reqConf = this.requestConfig(options);
-        reqConf.method = options.method || 'GET';
-        this.request(reqConf, callback);
+    constructor(attributes, options) {
+        super(attributes, options);
+
+        this.logger = hmpoLogger.get();
+        this._request = require('request');
     }
 
-    save(options, callback) {
-        if (typeof options === 'function' && arguments.length === 1) {
-            callback = options;
-            options = {};
-        } else if (!options) {
-            options = {};
-        }
+    fetch(callback) {
+        let config = this.requestConfig({method: 'GET'});
+        this.request(config, callback);
+    }
+
+    save(callback) {
         this.prepare((err, data) => {
 
-            if (err) {
-                return callback(err);
-            }
+            if (err) { return callback(err); }
 
             data = JSON.stringify(data);
 
-            let reqConf = this.requestConfig(options);
-
-            reqConf.method = options.method || 'POST';
+            let config = this.requestConfig({
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json',
+                    'Content-Length': Buffer.byteLength(data)
+                }
+            });
 
-            reqConf.headers = _.extend({
-                'Content-Type': 'application/json',
-                'Content-Length': Buffer.byteLength(data)
-            }, reqConf.headers || {});
+            this.request(config, data, callback);
 
-            this.request(reqConf, data, callback);
         });
     }
 
-    delete(options, callback) {
-        if (typeof options === 'function' && arguments.length === 1) {
-            callback = options;
-            options = {};
-        } else if (!options) {
-            options = {};
-        }
-        let reqConf = this.requestConfig(options);
-        reqConf.method = options.method || 'DELETE';
-        this.request(reqConf, callback);
+    delete(callback) {
+        let config  = this.requestConfig({method: 'DELETE'});
+        this.request(config, callback);
     }
 
-    requestConfig(options) {
-        let reqConf = this.url(options);
-        if (typeof reqConf === 'string') {
-            reqConf = url.parse(reqConf);
+    requestConfig(config) {
+        config.uri = this.url();
+
+        let auth = this.auth();
+
+        if (auth) {
+            config.auth = auth;
         }
-        return reqConf;
+
+        if (config.headers) {
+            config.headers = _.extend({}, config.headers);
```

Comment:
do we need to extend/"clone" these headers? the object passed from save/fetch etc is a new object each time and so could safely be mutated.